src/com/hms_networks/americas/sc/taginfo/TagInfoManager.java

```diff
@@ -534,6 +534,18 @@ private static boolean convertStrToBool(String stringBool) {
     return tagInfoList;
   }
 
+  /**
+   * Gets a {@link TagInfo} object from the tag info array populated by calling {@link
+   * #refreshTagList()}. If this method is called before {@link #refreshTagList()}, an {@link
+   * IllegalStateException} will be thrown.
+   *
+   * @param tagId the Ewon tag ID
+   * @return {@link TagInfo} object associated with the given tag ID
+   */
```

Comment:
The comments warn of an IllegalStateException. But no @throws? 
No throws for index out of range. What happens if tagID doesn't exist in TagInfoArray?

Reply:
This was missed, I will add this in

Reply:
if a tag Id does not exist it would throw an out of bounds exception if an index is out of bounds or null if you specify the index of a tag gap

Reply:
A new commit adds the missing throws exception to the old function that returns the array, and the commit with a new function to grab a single tag info object by id was updated to have the throws exceptions.